beacon/blockchain/finalize_block.go

```diff
@@ -116,7 +116,7 @@ func (s *Service) FinalizeSidecars(
 }
 
 func (s *Service) PostFinalizeBlockOps(ctx sdk.Context, blk *ctypes.BeaconBlock) error {
-	// TODO ABENEGIA: verify whether this is correct in the face of the extra copies
+	// TODO: consider extracting LatestExecutionPayloadHeader instead of using state here
```

Comment:
I don't like renegerating the state and I could cache lph way earlier and pass it down here.
On the other hand, we are reading the state so all should be correct here